Marlin/Marlin_main.cpp

```diff
@@ -6440,65 +6453,191 @@ inline void gcode_M17() {
     const millis_t ms = millis();
     if (ELAPSED(ms, next_buzz)) {
       if (max_beep_count < 0 || runout_beep < max_beep_count + 5) { // Only beep as long as we're supposed to
-        next_buzz = ms + ((max_beep_count < 0 || runout_beep < max_beep_count) ? 2500 : 400);
-        BUZZ(300, 2000);
+        next_buzz = ms + ((max_beep_count < 0 || runout_beep < max_beep_count) ? 1000 : 500);
+        BUZZ(880 - (runout_beep & 1) * 220, 50);
         runout_beep++;
       }
     }
   }
 
-  static void ensure_safe_temperature() {
-    bool heaters_heating = true;
+  static bool ensure_safe_temperature(const uint8_t &extruder=active_extruder, const AdvancedPauseMode mode=ADVANCED_PAUSE_MODE_PAUSE_PRINT) {
+
+    #if ENABLED(PREVENT_COLD_EXTRUSION)
+      if (!DEBUGGING(DRYRUN) && thermalManager.targetTooColdToExtrude(extruder)) {
+        SERIAL_ERROR_START();
+        SERIAL_ERRORLNPGM(MSG_HOTEND_TOO_COLD);
+        return false;
+      }
+    #endif
 
-    wait_for_heatup = true;    // M108 will clear this
-    while (wait_for_heatup && heaters_heating) {
+    wait_for_heatup = true; // M108 will clear this
+    while (wait_for_heatup && thermalManager.wait_for_heating(extruder)) {
       idle();
-      heaters_heating = false;
-      HOTEND_LOOP() {
-        if (thermalManager.degTargetHotend(e) && abs(thermalManager.degHotend(e) - thermalManager.degTargetHotend(e)) > TEMP_HYSTERESIS) {
-          heaters_heating = true;
-          #if ENABLED(ULTIPANEL)
-            lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_WAIT_FOR_NOZZLES_TO_HEAT);
+      #if ENABLED(ULTIPANEL)
+        lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_WAIT_FOR_NOZZLES_TO_HEAT, extruder, mode);
+      #endif
+    }
+
+    // Prompt user to load the filament, in this case
+    if (mode == ADVANCED_PAUSE_MODE_LOAD_FILAMENT) {
+      lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_INSERT, extruder, mode);
+      KEEPALIVE_STATE(PAUSED_FOR_USER);
+      filament_change_beep(max_beep_count, true);
+      wait_for_user = true;    // LCD click or M108 will clear this
+      while (wait_for_user) {
+        #if HAS_BUZZER
+          filament_change_beep(max_beep_count);
+        #endif
+      }
+    }
```

Comment:
@tcm0116 — I just added this block of code. During "Load" is the only time a prompt is needed.